### PR TITLE
PI-1533: Added staff caseload endpoint

### PR DESCRIPTION
**Pull request by pmcphee77**

### CODE REVIEW

File: projects/create-and-vary-a-licence-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/model/ManagedOffender.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package uk.gov.justice.digital.hmpps.api.model
+
+import java.time.LocalDate
+
+data class ManagedOffender(
+    val offenderCrn: String,
+    val offenderName: Name,
```
Comment (marcus-bcl) [inline projects/create-and-vary-a-licence-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/model/ManagedOffender.kt:7]: We probably don't need the `offender` prefix here
```suggestion
    val crn: String,
    val name: Name,
```

File: projects/create-and-vary-a-licence-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/resource/StaffResource.kt [hunk 2]
```diff
@@ -31,4 +32,9 @@ class StaffResource(
         staffService.findStaffForUsernames(
             usernames
         )
+
+    @PreAuthorize("hasRole('CVL_CONTEXT')")
+    @GetMapping("/{staffCode}/caseload/managed-offenders")
+    fun getManagedOffenders(@PathVariable staffCode: String): List<ManagedOffender> =
```
Comment (marcus-bcl) [inline projects/create-and-vary-a-licence-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/resource/StaffResource.kt:38]: The Community API version was paginated - do we need to do the same here?  Sorry if it's already been discussed.

Comment (pmcphee77) [reply projects/create-and-vary-a-licence-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/resource/StaffResource.kt:38]: The current community API is not currently paginated. See https://community-api-public.test.probation.service.justice.gov.uk/swagger-ui/index.html#/Staff/getCaseloadOffendersForStaff - i discussed this with Anthony and we decided not to introduce this.

File: projects/create-and-vary-a-licence-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/model/ManagedOffender.kt (hunk 1 above)
Comment (pmcphee77) [reply projects/create-and-vary-a-licence-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/model/ManagedOffender.kt]: Done

File: projects/create-and-vary-a-licence-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/resource/StaffResource.kt (hunk 2 above)
Comment (marcus-bcl) [reply projects/create-and-vary-a-licence-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/resource/StaffResource.kt:38]: My mistake - I was looking at the team caseload endpoint.

**Review by marcus-bcl — APPROVED** [review]

**Pull request merged**